Exiled.Events/Patches/Events/Map/DamagingWindow.cs

```diff
@@ -0,0 +1,37 @@
+// -----------------------------------------------------------------------
+// <copyright file="DamagingWindow.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Map
+{
+#pragma warning disable SA1313
+    using HarmonyLib;
+    using Mirror;
+
+    /// <summary>
+    /// Patches <see cref="BreakableWindow.ServerDamageWindow(float)"/>.
+    /// Adds the <see cref="Handlers.Map.DamagingWindow"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(BreakableWindow), nameof(BreakableWindow.ServerDamageWindow))]
+    internal static class DamagingWindow
+    {
+        private static bool Prefix(BreakableWindow __instance, ref float damage)
+        {
+            if (NetworkServer.active)
+            {
+                var ev = new EventArgs.DamagingWindowEventArgs(__instance, damage);
+                Handlers.Map.OnDamagingWindow(ev);
+                __instance.health -= ev.Damage;
+                if (__instance.health <= 0f)
+                {
+                    __instance.StartCoroutine(__instance.BreakWindow());
+                }
+            }
+
+            return false;

```

Comment:
Look at (for an example) here: [clickable](https://github.com/galaxy119/EXILED/blob/af4697b39c33b8a7158e3ed0183bab722653491b/Exiled.Events/Patches/Events/Map/AnnouncingDecontamination.cs)

You don't need to implement the whole method, it can just be a short event call